src/compartment.c

```diff
@@ -499,60 +595,60 @@ find_comp_by_addr(void* to_find)
  ******************************************************************************/
 
 // TODO WIP
-void
-setup_stack(struct Compartment* to_setup)
-{
-    assert(to_setup->stack_pointer % 16 == 0);
-
-    uintptr_t init_sp = to_setup->stack_pointer;
-    uintptr_t argv_ptrs[to_setup->argc];
-    for (size_t i = 0; i < to_setup->argc; ++i)
-    {
-        comp_stack_push(to_setup, to_setup->argv[i], strlen(to_setup->argv[i]));
-        argv_ptrs[i] = to_setup->stack_pointer;
-    }
-
-    uintptr_t envp_ptrs[ENV_FIELDS_CNT];
-    const char* envp_val;
-    for (size_t i = 0; i < ENV_FIELDS_CNT; ++i)
-    {
-        envp_val = get_env_str(comp_env_fields[i]);
-        comp_stack_push(to_setup, envp_val, strlen(envp_val));
-        envp_ptrs[i] = to_setup->stack_pointer;
-    }
-
-    size_t stack_push_size = (1 + 1 + 1 + sizeof(envp_ptrs) + sizeof(argv_ptrs)) * sizeof(uint64_t);
-    void* null_delim = NULL;
-    /* argc */
-    size_t stack_argc = to_setup->argc;
-    comp_stack_push(to_setup, &stack_argc, sizeof(uint64_t));
-    /* argv */
-    for (size_t i = 0; i < to_setup->argc; ++i)
-    {
-        comp_stack_push(to_setup, (void*) &argv_ptrs[i], sizeof(uint64_t));
-    }
-    /* argv NULL delimiter */
-    comp_stack_push(to_setup, &null_delim, sizeof(null_delim));
-    /* envp */
-    for (size_t i = 0; i < ENV_FIELDS_CNT; ++i) // envp
-    {
-        /*comp_stack_push(comp_envs[i], strlen(comp_envs[1] + 1);*/
-        comp_stack_push(to_setup, (void*) &envp_ptrs[i], sizeof(uint64_t));
-    }
-    /* envp NULL delimiter */
-    comp_stack_push(to_setup, &null_delim, sizeof(null_delim));
-    /* auxv */
-    comp_stack_auxval_push(to_setup, AT_PAGESZ, elf_aux_info(AT_PAGESZ, NULL, sizeof(size_t)));
-    comp_stack_auxval_push(to_setup, AT_PHDR, to_setup->phdr);
-    comp_stack_auxval_push(to_setup, AT_PHENT, to_setup->phentsize);
-    comp_stack_auxval_push(to_setup, AT_PHNUM, to_setup->phnum);
-    /*comp_stack_auxval_push(to_setup, AT_SECURE, 0);*/
-    /*comp_stack_auxval_push(to_setup, AT_RANDOM, rand());*/
-    comp_stack_auxval_push(to_setup, AT_NULL, 0);
-
-    to_setup->stack_pointer = init_sp;
-
-}
+/*void*/
```

Comment:
Commented code.

Reply:
This is also some tough code to remove; it was to do with stack setup when executing a binary, and it *might* be needed for argument passing to `main`, but I'm still unsure... I'd also like to keep this around for a bit longer.

Reply:
Code is never deleted from git history :) But if we really really think we are likely to need it soon, let's at least add a comment explaining why.

Reply:
Improved the TODO above the deleted block in https://github.com/capablevms/CHERI-ELF-comp/pull/6/commits/f1d99300bcb04db2753b1a18759f057b77fd1ac3.